Task3/Task3.c

```diff
@@ -0,0 +1,193 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#define BASESIZE 100
+#define LEN 100
+
+typedef struct
+{
+    char name[LEN];
+    char phone[LEN];
+} Entry;
+
+int readAndCountEntriesFromFile(Entry arrayOfEntries[], char* filename)
+{
+    int entriesCounter = 0;
+    FILE* file = fopen(filename, "a+");
+    if (file == NULL)
+    {
+        return -1;
+    }
+    while (!feof(file))
+    {
+        if ((fscanf(file, "%s", arrayOfEntries[entriesCounter].name) != EOF))
+        {
+            fscanf(file, "%s", arrayOfEntries[entriesCounter].phone);
+            entriesCounter++;
+        }
+    }
+    fclose(file);
+    return entriesCounter;
+}
+
+void begin()
+{
+    printf("\nEnter: \n 0 - to exit \n");
+    printf(" 1 - to add entry (name and phone number) \n");
+    printf(" 2 - to print all entries \n");
+    printf(" 3 - to find phone number by name \n");
+    printf(" 4 - to find name by phone number \n");
+    printf(" 5 - save current data to file\n");
+}
+
+void addEntry(Entry arrayOfEntries[], int* entriesCounter)
+{
+    if (*entriesCounter == BASESIZE)
+    {
+        printf("There's no room for new entries in the database! \n");
+    }
+    else
+    {
+        printf("Enter name (without spaces): ");
+        scanf("%s", &(arrayOfEntries[*entriesCounter].name));
+        printf("Enter phone (without spaces): ");
+        scanf("%s", &(arrayOfEntries[*entriesCounter].phone));
```

Comment:
```suggestion
        scanf("%s", arrayOfEntries[*entriesCounter].name);
        printf("Enter phone (without spaces): ");
        scanf("%s", arrayOfEntries[*entriesCounter].phone);
```
Строка - это массив, уже ссылочного типа, поэтому получать адрес на строку не нужно (следи за предупреждениями компилятора).